packages/nimble-components/src/wafer-map/modules/rendering.ts

```diff
@@ -5,41 +5,72 @@ import type { DataManager } from './data-manager';
  * Responsible for drawing the dies inside the wafer map
  */
 export class RenderingModule {
-    private readonly waferData: DataManager;
     private readonly context: CanvasRenderingContext2D;
+    private dieSize: number | undefined;
+    private readonly dies: DieRenderInfo[];
+    private readonly dimensions: Dimensions;
+    private readonly labelFontSize: number;
 
     public constructor(waferData: DataManager, canvas: HTMLCanvasElement) {
-        this.waferData = waferData;
         this.context = canvas.getContext('2d')!;
+        this.dies = waferData.diesRenderInfo;
+        this.dimensions = waferData.dieDimensions;
+        this.labelFontSize = waferData.labelsFontSize;
     }
 
-    public drawWafer(): void {
-        const dies: DieRenderInfo[] = this.waferData.diesRenderInfo;
-        const dimensions: Dimensions = this.waferData.dieDimensions;
+    public drawWafer(transform?: number): void {
+        this.renderDies(transform);
+        this.renderText(transform);
+    }
 
-        for (const die of dies) {
-            this.context.fillStyle = die.fillStyle;
-            this.context?.fillRect(
-                die.x,
-                die.y,
-                dimensions.width,
-                dimensions.height
-            );
+    public clearCanvas(width: number, height: number): void {
+        this.context.clearRect(0, 0, width, height);
+    }
 
-            this.context.font = this.waferData.labelsFontSize.toString();
-            this.context.fillStyle = '#ffffff';
-            this.context.textAlign = 'center';
-            const aproxTextHeight = this.context.measureText('M');
+    private renderDies(transform?: number): void {
+        this.dieSize = this.dimensions.width * this.dimensions.height * (transform || 1);
+        this.dies.sort((a, b) => a.fillStyle.localeCompare(b.fillStyle));

```

Comment:
I don't think we should do a locale aware comparison, seems like extra overhead. Can just do a manual check, ie a > b ret 1, a < b ret -1, else ret 0

mdn gives [perf advice for doing localized comparisons](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#performance) but we can skip the localization work altogether for fillStyle strings I assume